lib/ecto/changeset.ex

```diff
@@ -1217,6 +1217,22 @@ defmodule Ecto.Changeset do
     raise ArgumentError, "field names given to change/put_change must be atoms, got: `#{inspect(key)}`"
   end
 
+  defp put_change(data, changes, errors, valid?, key, new_value, {:parameterized, type, params}) do
+    old_value = Map.get(data, key)
+
+    case type.change(old_value, new_value, params) do
```

Comment:
You should use this branch for all types:

```suggestion
  defp put_change(data, changes, errors, valid?, key, new_value, type) do
    old_value = Map.get(data, key)

    case Ecto.Type.change(type, old_value, new_value) do
```

Reply:
Right, previously I was not sure if `Ecto.Type.change/3` would survive :-)